### ISSUE TITLE
API code needs to migrate from window.player

### ISSUE THREAD

**Issue by r2dev2** [LiveTL/LiveTL#338]
The api translation code is reliant upon `window.player` for `getDuration` but `window.player` is no longer accessible.

**Comment by GrumpyBear57** [comment]
I haven't touched this or anything else related to the UI, so I have no idea why you assigned this to me. It needs to be handled by someone who do UI work, probably by whoever is responsible for making `window.player` inaccessible.

**Comment by r2dev2** [comment]
@KentoNishi that's you

**Comment by ChrRubin** [comment]
For more information about this, `window.player` isn't always accessible even before Kento made the video player change. An example of this would be in Embedded or Popout mode, where the video player is never in the context of the script's window to begin with.

The only usage of `window.player` aside from the E2E testing scripts is the following:
https://github.com/LiveTL/LiveTL/blob/75354db254d662f1a9a9d64e5f9251ef36f77369/src/js/api.js#L99-L104

I'd assume this function would error out in Embedded or Popout mode in the current public version, but I can't test it since it only runs for live translations.

**Comment by ChrRubin** [comment]
Didn't want to necro #344, but I honestly still think we need a solution where live translations from API are in epoch timestamps, regardless of whether it is provided server-side or converted client-side. Live messages from YTC and mchad are already provided in epoch, and both the deduplication and spam protection code heavily depend on the messages all using the same timestamp format, which is currently epoch. 

If for whatever reason we keep trying to make live API translations stay in video time format, we would have to convert other sources to video time for deduplication to work, which at that point we might as well have just started with converting live API translations to epoch instead.

(Also btw the mchad script doesn't use the timestamp store at all, only the LTL API script uses it for archived translations)

**Comment by GrumpyBear57** [comment]
I was actually thinking about this yesterday, and while I still don't think
that having epoch/wall time as the single source of truth as to when the
message is for, it would make sense to have a field for the date/time the
message was created, for record keeping purposes (which I'm already doing
for modifications), which you could theoretically piggyback on for lives.
This is easy enough to do for the notifications (SSE) endpoint, but will
involve a bit more for persisting it. Do note that you'll only be able to
use this for lives, and vods will have to use the video timestamp, as well
as all created translations must still have a video timestamp, so that when
they become vods, it still works.

**Comment by ChrRubin** [comment]
Sounds like a good compromise. Sending TLs still won't work on external sites (especially with MultiView sites) unless they set up their own implementations for sending TLs to our API, but at least receiving TLs will theoretically work for all cases on the extension without having to fiddle with timestamp conversions.

**Comment by GrumpyBear57** [comment]
I've added the CreatedAt property to the notifications endpoint for now, which contains the time in UTC. Will finish adding it so that it persists later. 

Translation object now looks like this 
```json
{
    "id": 15838,
    "videoId": "M2gwJ-7s0Zo",
    "translatorId": "auth0|60c195aa5d89a500699c01cc",
    "languageCode": "en",
    "translatedText": "LTL API SSE Test",
    "start": 586292,
    "end": 591292,
    "state": "Created",
    "createdAt": "2021-10-05T03:17:12.122789Z"
}
```

I think this issue is also closable, as R2 fixed it in #344. Please use #276 for further API discussion.
